### DISCUSSION TITLE
pysindy-c

### DISCUSSION THREAD

**Discussion by anfh27** [dynamicslab/pysindy#443] [General]
Is the code for the paper:
https://www.sciencedirect.com/science/article/pii/S2405896316318298
here somewhere?
I'm interested in seeing the difference in the code for the forcing vs feedback control

**Comment by Jacob-Stevens-Haas** [comment]
Some of the code (forcing) is here.  You can check example 1 for the how-to.  Feedback control isn't present here - section 3 of that paper describes why it needs to be manually tweaked in order to fit unambiguously.  It might make sense that after system identification on perturbed control, simulation allowed feedback control via a callable on $x$ rather than $t$.

**Comment by anfh27** [comment]
If I am using already perturbed control input data, can I proceed with system identification with SINDy in the same way as usual for historical data, and also simulate for the holdout set in the usual way?

I would then write the callable for optimal feedback control for future data points - have I understood correctly?

**Reply by Jacob-Stevens-Haas** [reply]
Yes, you can proceed with system ID that way.

Sorry, I was pretty unclear above.  I was saying that it might make sense _as a feature request_ to allow control functions to operate on `x`, not `t`.  Not that pysindy allows it currently.
